src/modules/roc_fec/reader.cpp

```diff
@@ -162,6 +181,11 @@ void Reader::try_repair_() {
         return;
     }
 
+    if (!decoder_.begin(source_block_.size(), repair_block_.size())) {
+        alive_ = false;
```

Comment:
We should log the reason every time we're shutting down, e.g.:

```
        roc_log(LogDebug,
                "fec reader: can't begin decoder block, shutting down:"
                " sbl=%lu rbl=%lu", ...);
```

Also, we are we shutting down here? Shouldn't we skip a block instead?

Reply:
`begin` is usually failed when there is no enough memory. I am not sure if skipping a block or restarting a session will actually help us. I guess restarting a session will be more suitable because it requires more time and probably some memory will be released.

Reply:
My logic was the following: we can skip the "bad" block and try again next block, or next, or next.. If there are only a few bad blocks, we'll continue playing soon. And if not, watchdog will detect the silence and kill the session.

Reply:
Discussed to do it separately during #174.

Reply:
#179